### PR TITLE
[WIP]Changed autowrap to allow all iterables as the args parameter ( #10571 )

### PR DESCRIPTION
**Pull request by jerli**
I've attempted to address issue #10571. It should work, but let me know if there's a better approach.

### CODE REVIEW

**Comment by moorepants** [comment]
This will work for the specific case, but you need to make it work with any iterable, not just tuples. I'd suggest writing a few unit tests that check several types of iterables: list, tuple, set, etc.

**Comment by jerli** [comment]
Ah, oops, you're right. Thanks, my bad. 

I updated it, but since I believe the the list() function should work on all iterables, I decided to just check if args wasn't already a list, and then use list() to make it a list otherwise.

**Comment by moorepants** [comment]
You still must add some unit tests for this change. Check out the `tests` directory for examples.

**Comment by asmeurer** [comment]
There is an `iterable` function in sympy.core.compatibility that you should use here.

**Comment by jerli** [comment]
How is it now? Thanks for all the help, hopefully I'll get better at this soon.

**Comment by jksuom** [comment]
There are 6 queued builds in this branch. The older ones could probably be cancelled.

File: sympy/utilities/autowrap.py [hunk 1]
```diff
@@ -501,6 +501,7 @@ def autowrap(
 
     helpers = [helpers] if helpers else ()
     flags = flags if flags else ()
+    args = list(args) if iterable(args, exclude=list) else args
```
Comment (asmeurer) [inline sympy/utilities/autowrap.py]: You should use the default exclude.

Comment (jerli) [inline sympy/utilities/autowrap.py]: Alright, updated.

**Comment by jerli** [comment]
I just thought of an issue: when a set is passed as args, the order of args may not be maintained due to the nature of sets. Since order matters for the iterable in this function, should we discourage the use of sets as the args parameter? Or would there be any alternative recommendations?

**Comment by jerli** [comment]
I changed the commit so that sets can't be used, since I couldn't really think of any other worthwhile way to address this. Let me know if you guys have other suggestions.

**Pull request closed without merging**